### PR TITLE
Split src/import.ml into a stdlib

### PR DESCRIPTION
**Pull request by diml**
This PR splits the big `src/import.ml` into a new library `stdune`. The motivations are:
- cleanup a bit this module which is becoming big and where we dump a bit of everything
- so that we can split and share more things out of the `jbuilder` library, such as `Process`. This way the code for managing processes can be shared between `jbuilder` and `configurator`

I mostly just extracted every sub-module of Import into a new file in `src/stdune`. I took the opportunity to uniformize things a bit and changed a few things I was finding annoying:
- type `t` always come first
- `Map.map`, `Map.fold`, ... never pass the key to the callback while `Map.mapi`, `Map.foldi`, ... do
- removed the `~key` and `~data` labels, I find them useless and annoying
- `Set.elements` --> `Set.to_list`
- `Map.bindings` --> `Map.to_list`
- `Map.of_alist` --> `Map.of_list`
- added `Ordering.t` for comparison functions
- renamed `Inl/Inr` to `Left/Right`. The latter seems clearer
- moved `List.longest` to `String.longest`

I also added a new module `Stdune.Pp` for pretty-printers. It allows to decouple the pretty-printing from the actual rendering, which seems better. Currently we do the rendering at the point where we raise the exception, which is not ideal. I haven't made the change in this PR as it's already big, but eventually I'd like to change `Errors.Fatal_error` to something like:

```ocaml
exception Fatal_error of Colors.Style.t Pp.t
````

### CODE REVIEW

File: src/stdune/pp.mli [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+val char   : char   -> _ t
+val list   : ('a -> 'b t) -> 'a list -> 'b t
+
+val space   : _ t
+val cut     : _ t
+val newline : _ t
+
+val text : string -> _ t
+
+val tag : 'a -> 'a t -> 'a t
```
Comment (rgrinberg) [inline src/stdune/pp.mli]: No "`t` goes first" here?

**Comment by diml** [comment]
`js_of_ocaml` fails to build because of [this](https://github.com/ocsigen/js_of_ocaml/blob/master/lib/jbuild#L12). The glob is picking up `_utop.ml`, we need to rename it `_utop.ml-gen`.

File: src/stdune/pp.mli (hunk 1 above)
Comment (diml) [reply src/stdune/pp.mli]: True, I changed the order and also added a label. I think this API will change a bit more when we start using it anyway

**Comment by diml** [comment]
> Why were there changes required to bootstrap however?

The ordering of libraries was broken and it showed up in this PR, so I fixed it by using the proper top sort algorithm

**Comment by rgrinberg** [comment]
I see. Surprised that this didn't come up earlier.

**Pull request merged**